table/block_based/filter_policy.cc

```diff
@@ -100,18 +100,21 @@ class FastLocalBloomBitsBuilder : public XXH3pFilterBitsBuilder {
   ~FastLocalBloomBitsBuilder() override {}
 
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) override {
-    size_t num_entry = hash_entries_.size();
+    size_t num_entries = hash_entries_.size();
     std::unique_ptr<char[]> mutable_buf;
-    uint32_t len_with_metadata =
-        CalculateAndAllocate(num_entry, &mutable_buf, /*update_balance*/ true);
+    size_t len_with_metadata = CalculateAndAllocate(num_entries, &mutable_buf,
+                                                    /*update_balance*/ true);
 
     assert(mutable_buf);
     assert(len_with_metadata >= 5);
 
+    // Max size supported by implementation
+    assert(len_with_metadata <= 0xffffffffU);
+
     // Compute num_probes after any rounding / adjustments
-    int num_probes = GetNumProbes(num_entry, len_with_metadata);
+    int num_probes = GetNumProbes(num_entries, len_with_metadata);
 
-    uint32_t len = len_with_metadata - 5;
+    uint32_t len = static_cast<uint32_t>(len_with_metadata - 5);
```

Comment:
could `len` also be `size_t`?

Reply:
The filter implementation in bloom_impl.h uses FastRange32 to pick a 64-byte bucket. Although this could theoretically scale up to 64 << 32 bytes, the closer you get to that, the more skewed your distribution of elements into buckets can get. With a limit around 1 << 32, the skew is no more than ~2% (64 / 63 - 1).